### PR TITLE
Add test for Http Metadata exchange 

### PR DESCRIPTION
**Pull request by gargnupur**
**What this PR does / why we need it**:
Add test for Http Metadata exchange using network metadata exchange filter and prometheus stats filter.
This PR also, cleans up envoy config for http requests and makes it more easy to use and understand.

It currently depends on https://github.com/envoyproxy/envoy-wasm/pull/310 to work properly

### CODE REVIEW

File: src/envoy/tcp/metadata_exchange/metadata_exchange.cc [hunk 1]
```diff
@@ -257,7 +257,7 @@ void MetadataExchangeFilter::tryReadProxyData(Buffer::Instance& data) {
     setFilterState(config_->filter_direction_ == FilterDirection::Downstream
```
Comment (gargnupur) [inline src/envoy/tcp/metadata_exchange/metadata_exchange.cc]: Changes in this file are covered by https://github.com/istio/proxy/pull/2565/files PR

**Comment by gargnupur** [comment]
friendly ping... can I please get a review on this?
/cc @kyessenov , @bianpengyuan , @douglas-reid , @mandarjog

File: test/envoye2e/http_metadata_exchange/http_metadata_exchange_test.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+ "gcp_project": "test-project",
+ "gcp_cluster_location": "us-east4-b"
+},
+"LABELS": {
+ "app": "ratings",
+ "version": "v1",
+ "pod-template-hash": "84975bc778"
+},
+"ISTIO_PROXY_SHA": "istio-proxy:47e4559b8e4f0d516c0d17b233d127a3deb3d7ce",
+"NAME": "ratings-v1-84975bc778-pxz2w",`
```
Comment (kyessenov) [inline test/envoye2e/http_metadata_exchange/http_metadata_exchange_test.go]: nit: please don't embed config in sources. They are difficult to upgrade when the config needs to be changed.

Comment (gargnupur) [reply test/envoye2e/http_metadata_exchange/http_metadata_exchange_test.go]: Agreed.. there is a lot of common config in tests.. will update them all together into templates...

Comment (kyessenov) [reply test/envoye2e/http_metadata_exchange/http_metadata_exchange_test.go]: I started dumping them into https://github.com/istio/proxy/tree/master/testdata. You just need to assemble them as necessary.

Comment (gargnupur) [reply test/envoye2e/http_metadata_exchange/http_metadata_exchange_test.go]: will do that, in a cleanup cl following this PR

**Pull request merged**